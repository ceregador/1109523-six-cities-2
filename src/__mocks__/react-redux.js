export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
