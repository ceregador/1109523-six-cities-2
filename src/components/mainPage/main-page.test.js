import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import MainPage from './main-page.jsx';
import Renderer from 'react-test-renderer';
import CityPlaces from '../cityPlaces/city-places.jsx';
import EmptyCityPlaces from '../emptyCityPlaces/empty-city-places.jsx';
import CitiesList from '../citiesList/cities-list.jsx';
import PageHeader from '../pageHeader/page-header.jsx';

jest.mock(`../rentObjectCardList/rent-object-card-list.jsx`);
jest.mock(`../cityPlaces/city-places.jsx`);
jest.mock(`../citiesList/cities-list.jsx`);
jest.mock(`../emptyCityPlaces/empty-city-places.jsx`);
jest.mock(`../pageHeader/page-header.jsx`);
jest.mock(`react-redux`);

it(`renders CityPlaces and connects correctly`, () => {
  const tree = Renderer
    .create(
        <MainPage
          cities={[{name: `City`}]}
          isCityOffersExist={true}
          activeCity={{name: `City1`}}
          getOffers={() => undefined}
          changeActiveCity={() => null}
        />)
    .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({offers: []});
  expect(mappedProps).toHaveProperty(`cities`);
  expect(mappedProps).toHaveProperty(`activeCityName`);
  expect(mappedProps).toHaveProperty(`isCityOffersExist`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`getOffers`);
  expect(mapDispatchToProps).toHaveProperty(`changeActiveCity`);

  expect(connectAdvanced).toHaveBeenCalledWith(MainPage);

  expect(CityPlaces).toHaveBeenCalled();
  expect(CitiesList).toHaveBeenCalled();
  expect(PageHeader).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});

it(`renders EmptyCityPlaces and connects correctly`, () => {
  const tree = Renderer
    .create(
        <MainPage
          cities={[{name: `City`}]}
          isCityOffersExist={false}
          activeCity={{name: `City2`}}
          getOffers={() => undefined}
          changeActiveCity={() => null}
        />)
    .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(expect.any(Function), expect.any(Object));

  const mapStateToProps = connect.mock.calls[0][0];
  const mappedProps = mapStateToProps({offers: []});
  expect(mappedProps).toHaveProperty(`cities`);
  expect(mappedProps).toHaveProperty(`activeCityName`);
  expect(mappedProps).toHaveProperty(`isCityOffersExist`);

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`getOffers`);
  expect(mapDispatchToProps).toHaveProperty(`changeActiveCity`);

  expect(connectAdvanced).toHaveBeenCalledWith(MainPage);

  expect(EmptyCityPlaces).toHaveBeenCalled();
  expect(CitiesList).toHaveBeenCalled();
  expect(PageHeader).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
