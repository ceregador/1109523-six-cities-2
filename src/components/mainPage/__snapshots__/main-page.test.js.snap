// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CityPlaces and connects correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    />
    <div
      className="cities"
    />
  </main>
</div>
`;

exports[`renders EmptyCityPlaces and connects correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    />
    <div
      className="cities"
    />
  </main>
</div>
`;
